### PR TITLE
Scripts for releasing.

### LINKED ISSUE #510: Document/automate release process
There is a couple of steps to be done when doing a release, so it would be nice to have it documented and possibly automated. Steps that come to my mind are:

- bumping version of the `go.opentelemetry.io/otel` dependency in exporters and examples to `${NEW_VERSION}`
- updating the version string in `""go.opentelemetry.io/otel/sdk".Version()` function to `"${NEW_VERSION}"` (I discovered this function by accident yesterday and it still says "0.1.0"…)
- commit the changes
- file a PR with the changes
- after PR is merged, create git tags for otel-go and its submodules (`tag_latest_modules.sh` script does some of the work) and push them
  - alternatively, the commit with the changes and the created tags could be pushed directly to master as a single action if it's allowed
- update the github releases page

### PR DESCRIPTION
**Pull request by rghetia**
- fixes #510

It is not completely automated but two scripts allows to update go.mod and create tag for root/submodules.
- pre-release.sh - This updates go.mod
- tag.sh - Creates a new tag for root module and sub-modules.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Release Process
+
+## Pre-Release
+Update go.mod for submodules to depend on the new release which will happen
+in the next step. This will create build failure for those who depend
+on the master branch instead or released version. But they shouldn't be
+depending on the master. So it is not a concern.
+
+- Run
```
[the remark this answers is not in the document]
Comment (rghetia) [reply RELEASING.md]: Good point. fixed it.

File: pre_release.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+
+set -xe
+
+help()
+{
+   echo ""
+   echo "Usage: $0 -t tag"
+   echo "\t-t Unreleased tag. Update all go.mod with this tag."
```
[the remark this answers is not in the document]
Comment (rghetia) [reply pre_release.sh]: I add -e some places but left out others. Replaced with printf.

File: pre_release.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+   exit 1 # Exit script after printing help
+}
+
+while getopts "t:c:" opt
```
[the remark this answers is not in the document]
Comment (rghetia) [reply pre_release.sh]: yes, it was copy/paste error. removed it.

File: pre_release.sh [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+do
+   case "$opt" in
+      t ) TAG="$OPTARG" ;;
+      ? ) help ;; # Print help
+   esac
+done
+
+# Print help in case parameters are empty
+if [ -z "$TAG" ]
```
[the remark this answers is not in the document]
Comment (rghetia) [reply pre_release.sh]: yes. Added the check.

File: pre_release.sh [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+then
+   echo "Tag is missing";
+   help
+fi
+
+TAG_FOUND=`git tag --list ${TAG}`
+echo "TAG found $TAG_FOUND"
```
[the remark this answers is not in the document]
Comment (rghetia) [reply pre_release.sh]: it was just for debugging. removed it.

File: pre_release.sh [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+# Run lint to update go.sum
+make lint
+
+# Add changes and commit.
+git add .
+make ci
+git commit -m "Prepare for releasing $TAG"
+
+echo "Now run following to verify the changes.\ngit diff master"
+echo "\nThen push the changes to upstream"
```
[the remark this answers is not in the document]
Comment (rghetia) [reply pre_release.sh]: replaced with printf.

**Comment by rghetia** [comment]
@MrAlias PTAL.

File: tag.sh [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+	exit -1
+fi
+
+# Save most recent tag for generating logs
+TAG_CURRENT=`git tag | grep '^v' | tail -1`
+
+PACKAGE_DIRS=$(find . -mindepth 2 -type f -name 'go.mod' -exec dirname {} \; | egrep -v 'tools' | sed 's/^\.\///' | sort)
+
+# Create tag for root module
+git tag -a "${TAG}" -m "Version ${TAG}" ${COMMIT_HASH}
```
Comment (lizthegrey) [inline tag.sh]: For the future, you may want to sign the tag with `-s`

File: tag.sh [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+# Create tag for submodules
+for dir in $PACKAGE_DIRS; do
+	git tag -a "${dir}/${TAG}" -m "Version ${dir}/${TAG}" ${COMMIT_HASH}
```
Comment (lizthegrey) [inline tag.sh]: ditto here.

**Comment by lizthegrey** [comment]
We can delete https://github.com/open-telemetry/opentelemetry-go/blob/master/tag_latest_modules.sh now right?

File: tag.sh (hunk 7 above)
Comment (rghetia) [reply tag.sh]: opened #589 to update the script.

**Pull request merged**